ui/src/app/core/core/core.module.ts

```diff
@@ -0,0 +1,26 @@
+import { ModuleWithProviders, NgModule, Optional, SkipSelf } from '@angular/core';
+import { CommonModule } from '@angular/common';
+
+
+@NgModule({
+  declarations: [],
+  imports: [
+    CommonModule
+  ]
+})
+export class CoreModule {
+
+  constructor (@Optional() @SkipSelf() parentModule: CoreModule) {
+    if (parentModule) {
+      throw new Error(
+        'CoreModule is already loaded. Import it in the AppModule only');
+    }
+  }
+
+  static forRoot(): ModuleWithProviders<CoreModule> {
```

Comment:
why this function

Reply:
because it will call in import from the anther  module

Reply:
we do not need it just put the providers in @NgModule decorator